AccessControlException Failure in restricted environment

The JDJSTPjdbc testcase uses the JTOpen driver in a restricted environment.  Probably because of some JVM change, the following exception is now encountered. 

Exception in thread "main" java.lang.ExceptionInInitializerError
	at java.lang.J9VMInternals.ensureError(J9VMInternals.java:157)
	at java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:146)
	at com.ibm.as400.access.SocketContainerUnix2.setProperties(SocketContainerUnix2.java:97)
	at com.ibm.as400.access.PortMapper.getServerSocket(PortMapper.java:146)
	at com.ibm.as400.access.PortMapper.getServerSocket(PortMapper.java:129)
	at com.ibm.as400.access.AS400ImplRemote.signonConnect(AS400ImplRemote.java:3610)
	at com.ibm.as400.access.AS400ImplRemote.signon(AS400ImplRemote.java:3402)
	at com.ibm.as400.access.AS400.sendSignonRequest(AS400.java:3743)
	at com.ibm.as400.access.AS400.promptSignon(AS400.java:3298)
	at com.ibm.as400.access.AS400.signon(AS400.java:4669)
	at com.ibm.as400.access.AS400.connectService(AS400.java:1610)
	at com.ibm.as400.access.AS400.connectService(AS400.java:1567)
	at com.ibm.as400.access.AS400JDBCConnectionImpl.setProperties(AS400JDBCConnectionImpl.java:3519)
	at com.ibm.as400.access.AS400JDBCDriver.prepareConnection(AS400JDBCDriver.java:1474)
	at com.ibm.as400.access.AS400JDBCDriver.initializeConnection(AS400JDBCDriver.java:1308)
	at com.ibm.as400.access.AS400JDBCDriver.connect(AS400JDBCDriver.java:456)
	at com.ibm.as400.access.AS400JDBCDriver.connect(AS400JDBCDriver.java:265)
	at java.sql.DriverManager.getConnection(DriverManager.java:675)
	at java.sql.DriverManager.getConnection(DriverManager.java:258)
	at JDBCSample.main(JDBCSample.java:47)
Caused by: java.security.AccessControlException: Access denied ("java.util.PropertyPermission" "com.ibm.as400.access.native.library" "read")
	at java.security.AccessController.throwACE(AccessController.java:176)
	at java.security.AccessController.checkPermissionHelper(AccessController.java:238)
	at java.security.AccessController.checkPermission(AccessController.java:385)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:562)
	at java.lang.SecurityManager.checkPropertyAccess(SecurityManager.java:1307)
	at java.lang.System.getProperty(System.java:488)
	at java.lang.System.getProperty(System.java:471)
	at com.ibm.as400.access.NativeMethods.<clinit>(NativeMethods.java:53)
	... 18 more

The NativeMethods class needs to tolerate this exception and not rethrow it. 

Testcase recreate: 

java test.JDRunit 7586T JDJSTPjdbc 33,34,41

This error will only occur if running on the IBM i.

--- a/src/main/java/com/ibm/as400/access/NativeMethods.java
+++ b/src/main/java/com/ibm/as400/access/NativeMethods.java
@@ -13,6 +13,8 @@
 
 package com.ibm.as400.access;
 
+import java.security.AccessControlException;
+
 // The NativeMethods class is used to call the native methods for the IBM Toolbox for Java Native Classes.
 public class NativeMethods
 {
@@ -49,17 +51,21 @@ public class NativeMethods
 
         }
         if (Trace.traceOn_) Trace.log(Trace.DIAGNOSTIC, "Loading Native non-PASE methods ");  //@pase1
-
-        String alternateLibrary = System.getProperty("com.ibm.as400.access.native.library"); 
-        if (alternateLibrary != null) { 
-        	nativeLibraryQyjspart = "/QSYS.LIB/"+alternateLibrary+".LIB/QYJSPART.SRVPGM"; 
-            if (Trace.traceOn_) Trace.log(Trace.DIAGNOSTIC, "QYJSPART as "+nativeLibraryQyjspart);  
+        try{
+            String alternateLibrary = System.getProperty("com.ibm.as400.access.native.library"); 
+            if (alternateLibrary != null) { 
+                nativeLibraryQyjspart = "/QSYS.LIB/"+alternateLibrary+".LIB/QYJSPART.SRVPGM"; 
+                if (Trace.traceOn_) Trace.log(Trace.DIAGNOSTIC, "QYJSPART as "+nativeLibraryQyjspart);  
+            }
+        } catch(AccessControlException e)
+        {
+            Trace.log(Trace.DIAGNOSTIC, "Error checking property for custom native library:", e);
         }
         try{
             System.load(nativeLibraryQyjspart );  //if j9, then socket functions in this lib are overridden
         } catch(Throwable e)
         {
-                Trace.log(Trace.ERROR, "Error loading QYJSPART service program:", e); //may be that it is already loaded in multiple .war classloader
+            Trace.log(Trace.ERROR, "Error loading QYJSPART service program:", e); //may be that it is already loaded in multiple .war classloader
         }
     }
 

